src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -874,6 +923,456 @@ string M(object o, bool b)
+        [|var|] ae = e as C;
+        if (fe != null)
+        {
+            M(ae);
+        }
+        else if (ae != null)
+        {
+            M(ae);
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestUseBeforeDeclaration()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        {
+            {
+                var x1 = c;
+
+                if (c != null)
+                {
+
+                }
+            }
+        }
+    }
+
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestPossiblyUnassigned()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        {
+            {
+                if (c != null)
+                {
+
+                }
+
+                var x2 = c;
+            }
+
+            // out of scope
+            var x3 = c;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestOutOfScope()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        {
+            {
+                if (c != null)
+                {
+
+                }
+            }
+
+            // out of scope
+            var x3 = c;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestDeclarationOnOuterBlock()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        {
+            {
+                if (c != null)
+                {
+
+                }
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        {
+            {
+                if (e is C c)
+                {
+
+                }
+            }
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestConditionalExpression()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|var|] c = e as C;
+        M(c != null ? c : null);
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        M(e is C c ? c : null);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestConditionalExpression_OppositeBranch()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e, C x)
+    {
+        [|var|] c = e as C;
+        M(c != null ? c : null, c);
```

Comment:
>M(c != null ? c : null, c); [](start = 8, length = 27)

I didn't understand why the declaration of `c` can't be inlined here. 
Is it a stylistic reason? I'm not sure I'd encourage folks to write `M(e is C c ? c : null, c);` ;-)

Maybe the test name should be updated?